src/main/java/com/google/neighborgood/servlets/TaskServlet.java

```diff
@@ -93,8 +96,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       }
       out.append("<div class='task-container'>");
       out.append("<div class='task-header'>");
+
+      // Retrieves task's user to get user's nickname
+      String taskOwner = (String) entity.getProperty("Owner");
+      Query userQuery =
+          new Query("UserInfo")
+              .setFilter(new FilterPredicate("userId", FilterOperator.EQUAL, taskOwner));
+      PreparedQuery userResults = datastore.prepare(userQuery);
+      Entity userEntity = userResults.asSingleEntity();
+
+      if (userEntity == null) {
+        System.err.println("Unable to find the user entity based on the current user id");
+        response.sendError(
+            HttpServletResponse.SC_NOT_FOUND, "The requested user could not be found");
+        return;
+      }
+
       out.append("<div class='username'>")
```

Comment:
Might be worth changing this class name in we're showing nicknames now, instead of usernames. Once you merge your CSS rules, it's possible you'll want to display usernames and nicknames differently, in which case this could get displayed in an inconsistent way.

Reply:
This is a good idea, I'll change this to nicknames, thanks! Should we merge the CSS files into one for our project or keep each page with its own CSS file? I looked up online what the typical approach was but found conflicting answers so I wasn't sure if we should merge them or no at the end.

Reply:
I suspect you'll have some similarities across pages that you want to share, so one option would be to have a `base.css` file and then keep page-specific ones. Then have every page use the base CSS alongside its own file. That can of course happen in another PR.

Reply:
That makes sense, thanks for clarifying!